sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala

```diff
@@ -344,4 +345,16 @@ object CommandUtils extends Logging {
   private def isDataPath(path: Path, stagingDir: String): Boolean = {
     !path.getName.startsWith(stagingDir) && DataSourceUtils.isDataPath(path)
   }
+
+  def getSizeInBytesFallBackToHdfs(session: SparkSession, path: Path, defaultSize: Long): Long = {
+    try {
+      val hadoopConf = session.sessionState.newHadoopConf()
+      val fs: FileSystem = path.getFileSystem(hadoopConf)
+      fs.getContentSummary(path).getLength
+    } catch {
+      case e: IOException =>
+        logWarning("Failed to get table size from hdfs. Returning default sizeInBytes")
```

Comment:
Shall we add the real value `$defaultSize` to the message, too?

BTW, we are using two formulas. The first one is not `sizeInBytes`. So, please revise the message in a more neutral way. Maybe, something like `Using the default size, ${defaultSize}`?
- `HadoopFsRelation` is using `(location.sizeInBytes * compressionFactor).toLong`.
- `DetermineTableStats` is using `session.sessionState.conf.defaultSizeInByte`.

Reply:
+1 for adding `$defaultSize` in the msssage.

Reply:
Added $defaultSize in the warning message. Thanks